src/components/themed/WalletListHeader.js

```diff
@@ -34,27 +39,85 @@ type DispatchProps = {
 type Props = OwnProps & StateProps & DispatchProps & ThemeProps
 
 type State = {
-  sorting: boolean,
-  showSlidingTutorial: boolean
+  input: string
 }
 
 class WalletListHeaderComponent extends React.PureComponent<Props, State> {
+  textInput = React.createRef()
+
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      input: ''
+    }
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.searching === false && this.props.searching === true && this.textInput.current) {
+      this.textInput.current.focus()
+    }
+  }
+
   activateSorting = () => this.props.toggleSorting(true)
 
+  handleOnChangeText = (input: string) => this.props.onChangeSearchText(input)
+
+  handleTextFieldFocus = () => {
+    this.props.toggleWalletSearching(true)
+  }
+
+  disabledTextFieldFocus = () => {
+    this.clearText()
+    this.props.toggleWalletSearching(false)
+  }
+
+  clearText = () => {
+    this.setState({ input: '' })
+    // $FlowFixMe - react-native-material-textfield have many flow errors. Somehow needed cause material-textfield value is not functioning well
+    this.textInput.current.clear()
+    this.props.onChangeSearchText('')
+  }
+
   render() {
-    const { sorting, theme } = this.props
+    const { sorting, searching, theme } = this.props
     const styles = getStyles(theme)
 
     return (
       <>
-        <WiredBalanceBox
-          showBalance={getIsAccountBalanceVisible}
-          fiatAmount={getTotalFiatAmountFromExchangeRates}
-          isoFiatCurrencyCode={getDefaultIsoFiat}
-          onPress={this.props.toggleAccountBalanceVisibility}
-          exchangeRates={this.props.exchangeRates}
-        />
-        {!sorting && (
+        <View style={styles.searchContainer}>
+          <View style={{ flex: 1, flexDirection: 'column' }}>
+            <EdgeTextFieldOutlined
+              returnKeyLabel={s.strings.fragment_seach}
+              returnKeyType="done"
+              label={s.strings.wallet_list_wallet_search}
+              onChangeText={this.handleOnChangeText}
+              value={this.state.input}
+              onFocus={this.handleTextFieldFocus}
+              ref={this.textInput}
+              marginRem={0}
+            />
+          </View>
+          {searching && (
+            <>
+              <TouchableOpacity onPress={this.clearText} style={styles.searchClearIcon}>
```

Comment:
This icon should be inside of the outlined text field. This looks like a bad hack. Create a new component OutlinedTextFieldWithClear that always has the X icon on the right side.